packages/swapper/src/swappers/ButterSwap/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,232 @@
+        code: TradeQuoteError.UnsupportedChain,
+      }),
+    )
+  }
+
+  const feeAssetId = chainIdToFeeAssetId(sellAsset.chainId)
+  const sellAssetIsNative = sellAsset.assetId === feeAssetId
+  const buyAssetIsNative = buyAsset.assetId === chainIdToFeeAssetId(buyAsset.chainId)
+
+  const { assetReference: sellAssetAddressRaw } = fromAssetId(sellAsset.assetId)
+  const { assetReference: buyAssetAddressRaw } = fromAssetId(buyAsset.assetId)
+
+  const sellAssetAddress = sellAssetIsNative ? ZERO_ADDRESS : sellAssetAddressRaw
+  const buyAssetAddress = buyAssetIsNative ? ZERO_ADDRESS : buyAssetAddressRaw
+  const slippageDecimal =
+    slippageTolerancePercentageDecimal ??
+    getDefaultSlippageDecimalPercentageForSwapper(SwapperName.ButterSwap)
+  const slippage = bn(slippageDecimal).times(10000).toString()
+
+  // Call ButterSwap /route API
+  const routeResult = await getRoute(
+    butterSwapFromChainId,
+    sellAssetAddress,
+    butterSwapToChainId,
+    buyAssetAddress,
+    bn(amount).shiftedBy(-sellAsset.precision).toString(), // convert to human units
+    slippage,
+  )
+
+  if (routeResult.isErr()) return Err(routeResult.unwrapErr())
+  const routeResponse = routeResult.unwrap()
+  if (!isRouteSuccess(routeResponse)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeQuote] ${routeResponse.message}`,
+        code: TradeQuoteError.QueryFailed,
+      }),
+    )
+  }
+
+  const route = routeResponse.data[0]
+  if (!route) {
+    return Err(
+      makeSwapErrorRight({
+        message: '[getTradeQuote] No route found',
+        code: TradeQuoteError.NoRouteFound,
+      }),
+    )
+  }
+
+  // Call ButterSwap /swap API to get calldata and contract info
+  const buildTxResult = await getBuildTx(
+    route.hash,
+    slippage,
+    sendAddress, // from (source chain address)
+    receiveAddress, // receiver (destination chain address)
+  )
+
+  if (buildTxResult.isErr()) return Err(buildTxResult.unwrapErr())
+  const buildTxResponse = buildTxResult.unwrap()
+  if (!isBuildTxSuccess(buildTxResponse)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeQuote] /swap failed: ${buildTxResponse.message}`,
+        code: TradeQuoteError.QueryFailed,
+      }),
+    )
+  }
+  const buildTx = buildTxResponse.data[0]
+  if (!buildTx) {
+    return Err(
+      makeSwapErrorRight({
+        message: '[getTradeQuote] No buildTx data returned',
+        code: TradeQuoteError.QueryFailed,
+      }),
+    )
+  }
+
+  // Fee asset for network/protocol fees
+  const feeAsset = _deps.assetsById[feeAssetId]
+  if (!feeAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeQuote] Fee asset not found for chainId ${sellAsset.chainId}`,
+        code: TradeQuoteError.UnsupportedChain,
+      }),
+    )
+  }
+
+  // Map gasFee.amount to networkFeeCryptoBaseUnit using fee asset precision
+  const networkFeeCryptoBaseUnit = bnOrZero(route.gasFee?.amount).gt(0)
```

Comment:
is this correct? This looks pessimistic

1. e.g when I input that I want to sell 0.01 ETH on butter, the quote I get is for 0.01 ETH - `gasFee` and in reality, the amount that gets swapped is not 0.01 ETH, but 0.009999 ETH. Looks like we'll want to factor that in somehow (interestingly, the rate still seems better than other swappers despite that)

<img width="312" alt="image" src="https://github.com/user-attachments/assets/25418b4f-c238-4071-b6d0-bf5b29f147ec" />